### PR TITLE
432 groups export

### LINKED ISSUE #432: Discount Code Export: add Code Groups
### Description
Discount Code Groups were recently added to the API. In order for customers to appropriately export the group information, we need to provide the possibility to export groups within the discount code export. 

API documentation: https://dev.commercetools.com/http-api-projects-discountCodes.html#discountcode

FYI @ValeSauer

### PR DESCRIPTION
**Pull request by wizzy25**
#### Summary

Support exporting of (discount-code)[https://dev.commercetools.com/http-api-projects-discountCodes.html#discountcode] groups in CSV

Also contains minor refactorings to make code more readable, and removes the redundant `index.js` file

resolves #432

### CODE REVIEW

File: packages/discount-code-exporter/src/main.js [hunk 1]
```diff
@@ -137,17 +137,23 @@ export default class DiscountCodeExport {
...
+    const { cartDiscounts, groups, ...restCodeData } = data
+    const cartDiscountsString = cartDiscounts.reduce(
       (acc: string, discount: Object): string => {
         if (!acc) return discount.id
         return `${acc}${this.config.multiValueDelimiter}${discount.id}`
       },
       ''
     )
+    const groupsString = groups
+      ? groups.reduce((acc: string, group: string): string => {
```
Comment (tdeekens) [inline packages/discount-code-exporter/src/main.js]: Whenever I read a `reduce` with `acc` I wonder what `acc` actually is. In our case `joinedGroups`?

File: packages/discount-code-exporter/src/main.js [hunk 2]
```diff
@@ -137,17 +137,23 @@ export default class DiscountCodeExport {
     }).discountCodes
   }
 
-  // Use this method to make the `cartDiscounts`
+  // Use this method to make the `cartDiscounts` and `groups`
   // field compatible with the importer
   _processCode(data: CodeData): Object {
-    const newCodeObj = data
-    const cartDiscounts = newCodeObj.cartDiscounts.reduce(
+    const { cartDiscounts, groups, ...restCodeData } = data
```
Comment (tdeekens) [inline packages/discount-code-exporter/src/main.js]: `remainingDiscountCodeData`?

File: packages/discount-code-exporter/src/main.js [hunk 3]
```diff
@@ -137,17 +137,23 @@ export default class DiscountCodeExport {
...
       (acc: string, discount: Object): string => {
         if (!acc) return discount.id
         return `${acc}${this.config.multiValueDelimiter}${discount.id}`
       },
       ''
     )
+    const groupsString = groups
+      ? groups.reduce((acc: string, group: string): string => {
+          if (!acc) return group
+          return `${acc}${this.config.multiValueDelimiter}${group}`
```
Comment (adnasa) [inline packages/discount-code-exporter/src/main.js]: wouldn't it have been more clearer to use a mapper + join?

File: packages/discount-code-exporter/src/main.js (hunk 1 above)
Comment (wizzy25) [reply packages/discount-code-exporter/src/main.js]: I just have the `acc` name stuck in my head no matter what 😄

File: packages/discount-code-exporter/src/main.js [hunk 4]
```diff
@@ -164,12 +157,15 @@ export default class DiscountCodeExport {
       'customLineItemFieldTypes',
     ]
     objKeys.forEach((key: string) => {
-      if (restCodeData[key] && !Object.keys(restCodeData[key]).length)
-        delete restCodeData[key]
+      if (
+        restDiscountCodeData[key] &&
+        !Object.keys(restDiscountCodeData[key]).length
+      )
+        delete restDiscountCodeData[key]
```
Comment (adnasa) [inline packages/discount-code-exporter/src/main.js]: nooo I didn't see this earlier 😭  #mutation 
wouldn't it be cooler to use `entries` + `reduce` here?

at the top of my head: 

```
const objKeys = [
  'attributeTypes',	      
  'cartFieldTypes',	       
  'lineItemFieldTypes',	     
  'customLineItemFieldTypes',
]

discountCodeData = Object.entries(restDiscountCodeData).reduce(
  (discountCode, ([discountCodeKey, value])) =>
    objKeys.includes(discountCodeKey)
      ? {
        ...discountCode,
        [discountCodeKey]: undefined
      } : discountCode,
  {}
)
```

File: packages/discount-code-exporter/src/main.js [hunk 5]
```diff
@@ -156,19 +156,18 @@ export default class DiscountCodeExport {
...
+
+    const discountCodeData = Object.entries(restDiscountCodeData).reduce(
+      (discountCode, [discountCodeKey, value]) =>
+        objKeys.includes(discountCodeKey) && !Object.keys(value).length
+          ? discountCode
+          : { ...discountCode, [discountCodeKey]: value },
+      {}
     )
+    return flatten({
+      ...discountCodeData,
```
Comment (adnasa) [inline packages/discount-code-exporter/src/main.js]: aaah, that fresh breeze of object-transforms! 
ooh so fresh! 🌬

**Pull request merged**